pkg/client/cache/index.go

```diff
@@ -28,6 +28,8 @@ type Indexer interface {
 	Store
 	// Retrieve list of objects that match on the named indexing function
 	Index(indexName string, obj interface{}) ([]interface{}, error)
+	// Retrieve list of all index names
+	ListIndexNames(name string) []string
```

Comment:
When PVs get indexed, their access modes are converted into a string and that string is the index.   Different modes render a different string.

A user makes a PVClaim with some number of access modes that might not match the indexed string (e.g, they request just ReadWriteMany.   This matches NFS (RWO+ROX+RWX).    

I asked the index for all "accessmodes" (the name arg) and from there I get the keys in the index, which are string representations of the access modes of all volumes.    I can then find a RWX mode among the many keys and give the user a match.

persistent_volume_index_test.go, TestFindingVolumeWithDifferentAccessModes is the main test case for this PR.